google/cloud/pubsublite/internal/default_routing_policy_test.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsublite/internal/default_routing_policy.h"
+#include <gmock/gmock.h>
+#include <nlohmann/json.hpp>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+namespace {
+
+using google::cloud::pubsublite_internal::DefaultRoutingPolicy;
+
+TEST(DefaultRoutingPolicyTest, RouteWithKey) {
+  std::string data = R"""({
+  "oaisdhfoiahsd": 18,
```

Comment:
If you're not loading the JSON file, instead just use std::unordered_map<std::string, int> directly